sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -764,3 +764,25 @@ def _append_original_pr_name_env(task_template):
     for step in task_template['taskSpec']['steps']:
         if step['name'] == 'main':
             _append_original_pr_name_env_to_step(step)
+
+
+def fix_big_data_passing_using_volume(workflow, pipeline_conf):
+    if workflow['spec'].get('workspaces'):
+        if pipeline_conf.data_passing_method._volume != None:
```

Comment:
It's quite difficult to read in this nested form. How about borrowing a Go idiom and go with:
```python
if pipeline_conf.data_passing_method._volume is None:
  return workflow
```
etc?